datacompy/polars.py

```diff
@@ -0,0 +1,984 @@
+
+        if min(sample_count, self.df2_unq_rows.shape[0]) > 0:
+            report += (
+                f"Sample Rows Only in {self.df2_name} (First {column_count} Columns)\n"
+            )
+            report += (
+                f"---------------------------------------{'-' * len(self.df2_name)}\n"
+            )
+            report += "\n"
+            columns = self.df2_unq_rows.columns[:column_count]
+            unq_count = min(sample_count, self.df2_unq_rows.shape[0])
+            report += df_to_str(self.df2_unq_rows.sample(unq_count)[columns])
+            report += "\n\n"
+
+        if html_file:
+            html_report = report.replace("\n", "<br>").replace(" ", "&nbsp;")
+            html_report = f"<pre>{html_report}</pre>"
+            with open(html_file, "w") as f:
+                f.write(html_report)
+
+        return report
+
+
+def render(filename: str, *fields: Union[int, float, str]) -> str:
+    """Renders out an individual template.  This basically just reads in a
+    template file, and applies ``.format()`` on the fields.
+
+    Parameters
+    ----------
+    filename : str
+        The file that contains the template.  Will automagically prepend the
+        templates directory before opening
+    fields : list
+        Fields to be rendered out in the template
+
+    Returns
+    -------
+    str
+        The fully rendered out file.
+    """
+    this_dir = os.path.dirname(os.path.realpath(__file__))
+    with open(os.path.join(this_dir, "templates", filename)) as file_open:
+        return file_open.read().format(*fields)
+
+
+def columns_equal(
+    col_1: "pl.Series",
+    col_2: "pl.Series",
+    rel_tol: float = 0,
+    abs_tol: float = 0,
+    ignore_spaces: bool = False,
+    ignore_case: bool = False,
+) -> "pl.Series":
+    """Compares two columns from a dataframe, returning a True/False series,
+    with the same index as column 1.
+
+    - Two nulls (np.nan) will evaluate to True.
+    - A null and a non-null value will evaluate to False.
+    - Numeric values will use the relative and absolute tolerances.
+    - Decimal values (decimal.Decimal) will attempt to be converted to floats
+      before comparing
+    - Non-numeric values (i.e. where np.isclose can't be used) will just
+      trigger True on two nulls or exact matches.
+
+    Parameters
+    ----------
+    col_1 : Polars.Series
+        The first column to look at
+    col_2 : Polars.Series
+        The second column
+    rel_tol : float, optional
+        Relative tolerance
+    abs_tol : float, optional
+        Absolute tolerance
+    ignore_spaces : bool, optional
+        Flag to strip whitespace (including newlines) from string columns
+    ignore_case : bool, optional
+        Flag to ignore the case of string columns
+
+    Returns
+    -------
+    Polars.Series
+        A series of Boolean values.  True == the values match, False == the
+        values don't match.
+    """
+    compare: pl.Series
+    try:
```

Comment:
For this function, should we check the type first before trying matching logic? I.e. Something like [this check](https://docs.pola.rs/py-polars/html/reference/series/api/polars.Series.is_numeric.html) to check if the first column is numeric?

Reply:
Good call. I might tweak that a bit. I just took the existing pandas version and ported it.

Reply:
@ak-gupta something like.. ?

```python
    if col_1.is_numeric() and col_2.is_numeric():
        compare = pl.Series(
            np.isclose(col_1, col_2, rtol=rel_tol, atol=abs_tol, equal_nan=True)
        )
    else:
        try:
```

Reply:
Yeah! I think doing those checks might make the code a bit more efficient. Similarly, we could use the `is_temporal` check for datetime columns as well.

Reply:
Maybe something like

```python
if col_1.dtype.is_numeric() and col_1.dtype.is_numeric():
    # Numeric check with float cast
   return ...

either_temporal: bool = col_1.dtype.is_temporal() or col_2.dtype.is_temporal()
for col in [col_1, col_2]:
    if str(col.dtype) in STRING_TYPE:
        # Ignore case and spaces
        if either_temporal:
            # Cast to datetime

if either_temporal:
    # Datetime comparison
else:
    # eq_missing comparison
```

Given that your current logic is a lift-and-shift implementation from Pandas we can mark the update as optional.